assets/javascripts/modules/youtubePlayer.js

```diff
@@ -0,0 +1,132 @@
+ *
+ * Usage example:
+
+    <div class='youtube-player-container'>
+       <div id='tax-help' class='js-youtube-player' data-video-id='MlMnz0Omosk' data-ga-play-event='tax-return-help:Click:play'></div>
+    </div>
+
+
+ * Links decorated with js-youtube-skip class will make the player seek ahead to the time given and start.
+ * data-player-id is required to identify which player is to be controlled.
+ * Usage example:
+
+  <a class='js-youtube-skip' href='#' data-player-id="tax-help" data-skip-time='120'>Two minutes later</a>
+
+ */
+
+
+var $youTubePlayerElems;
+var YoutubePlayers;
+
+/**
+ * Called by Youtube script to create players on page. Needs to be set on the root object
+ */
+window.onYouTubeIframeAPIReady = function() {
+
+  $youTubePlayerElems.each(function() {
+    elementId = $(this).attr('id');
+    //player = YoutubePlayer[elementId] = {};
+
+    YoutubePlayers[elementId].player = new YT.Player(
+      elementId, {
+        videoId: YoutubePlayers[elementId].params.videoId,
+        events: {
+          'onStateChange': onPlayerStateChange
+        }
+      });
+  });
+};
+
+
+/**
+ * Fires Google Analytics event via GDS Stageprompt library.
+ * Event values provided by data-ga-play-event attribute on player containing element.
+ * @param event
+ */
+function onPlayerStateChange(event) {
+
+  var playerId = event.target.getIframe().id;
+
+  if (YoutubePlayers[playerId] && !YoutubePlayers[playerId].started && event.data === YT.PlayerState.PLAYING) {
+    YoutubePlayers[playerId].started = true;
+    var gaEventFields = YoutubePlayers[playerId].params.gaEvent.split(':');
+    if (gaEventFields.length === 3) {
+      GOVUK.performance.sendGoogleAnalyticsEvent(gaEventFields[0], gaEventFields[1], gaEventFields[2]);
+    }
+  }
+}
+
+/**
+ * Find all players on page and initialise them from the data attribute values
+ */
+var initialisePlayers = function() {
+  var elementId;
+  var player;
+
+  YoutubePlayers = {};
+
+  $youTubePlayerElems.each(function(index, elem) {
+      elementId = $(this).attr('id');
+      player = YoutubePlayers[elementId] = {};
+
+      player.started = false;
+      player.params = {
+        elementId: elementId,
+        videoId: $(this).data('video-id') || '',
+        gaEvent: $(this).data('ga-play-event') || ''
+      };
+    }
+  );
+};
+
+/**
+ * Brings in JS API script from Youtube.
+ */
+var addYouTubeAPIScript = function() {
+  var tag = document.createElement('script');
+  tag.src = 'https://www.youtube.com/iframe_api';
+  var firstScriptTag = document.getElementsByTagName('script')[0];
+  firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+};
+
+/**
+ * if players present on page, initialise and bring in API script
+ */
+var init = function() {
+  $youTubePlayerElems = $('.js-youtube-player');
+
+  if ($youTubePlayerElems.length) {
+    initialisePlayers();
+    addYouTubeAPIScript();
+  }
+};
+
+$('.js-youtube-skip').click(function(event) {
```

Comment:
feels like this could be in a

```
var addListeners = function(){
...
}
```
and `$('.js-youtube-skip')` could have the same treatment as `var $youTubePlayerElems;`